src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size;
+    private Node head;
+    private Node tail;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> node = new Node<>(value, null, null);
+        if (isEmpty()) {
+            linkHead(node);
+            linkTail(node);
+            size++;
+            return true;
+        }
+        tail.next = node;
+        node.prev = tail;
+        linkTail(node);
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        Node<T> node = new Node(value, null, null);
+        if (index == 0) {
+            if (isEmpty()) {
+                add(value);
+                return;
+            }
+            head.prev = node;
+            node.next = head;
+            linkHead(node);
+            size++;
+            return;
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        node.next = getNode(index);
+        node.prev = getNode(index - 1);
+        getNode(index).prev = node;
+        getNode(index - 1).next = node;
```

Comment:
not to search your list 4 times, call getNode() once and save result to sum temp variable